numalogic/udfs/tools.py

```diff
@@ -107,6 +121,26 @@ def _load_model(
             payload.composite_keys,
             payload.metrics,
         )
-        return None
+        return None, payload
     else:
-        return artifact
+        _LOGGER.info(
+            "%s - Loaded Model. Source: %s , version: %s, Keys: %s, %s",
+            payload.uuid,
+            artifact.extras.get("source"),
+            artifact.extras.get("version"),
+            skeys,
+            dkeys,
+        )
+        if (
```

Comment:
add a TODO here to not mutate the payload here

Reply:
Sure